### PR TITLE
Add GitHub Actions as new CI

### PR DESCRIPTION
**Pull request by volpatto**
Self-explained title.

### CODE REVIEW

File: .github/workflows/win.yml
Comment (volpatto) [inline .github/workflows/win.yml:40]: I added this for debug purposes only. I will remove as soon as we have Win build fixed.

**Comment by volpatto** [comment]
@nicoddemus sorry for the trouble, but do you have any idea how to solve Windows builds? It looks like a Path issue.

File: .github/workflows/win.yml
Comment (prusse-martin) [reply .github/workflows/win.yml:40]: The value displayed for **PATH** env var makes it look like the env vars set by conda devenv are not taking hold. Maybe this is a problem with a new conda on windows.

You could try to disable `auto-update-conda` and force a old version of conda (4.8.3 comes to mind). If it works we could narrow down the latest version where it works... If is a bug we will need to fix it on conda-devenv or conda itself.

Comment (volpatto) [reply .github/workflows/win.yml:39]: Hmmmm! Great, @prusse-martin! Thanks a lot, I will test some ways to use an older version of conda.

Comment (volpatto) [reply .github/workflows/win.yml:39]: Pinned to `4.8.3`, no luck so far.

File: README.rst
Comment (nicoddemus) [inline README.rst:9]: We should add a badge for GH actions here (see pytest-regressions for an example).

File: .github/workflows/win.yml
Comment (nicoddemus) [reply .github/workflows/win.yml:39]: Strange... try adding this after `conda activate`:

```
echo PYTHONPATH=%PYTHONPATH%
where python
```

File: README.rst
Comment (volpatto) [reply README.rst:9]: Done! Thanks!

Comment (nicoddemus) [reply README.rst:9]: Wait, is this powershell? That might explain why the activate is not working.

Comment (volpatto) [reply README.rst:9]: Ah, yes! It is powershell, sorry... so, there is a workaround? Why activate is not working? (sorry for the dumb questions, I'm not experienced with Windows)

Comment (volpatto) [reply README.rst:9]: I see... there is no `PYTHONPATH`...

Comment (volpatto) [reply README.rst:9]: We can change the `shell`... [here](https://github.com/marketplace/actions/setup-miniconda) some examples are provided

Comment (nicoddemus) [reply README.rst:9]: Yeah Martin is right, activate is not working.

Try `conda init powershell` before `conda activate`.

File: .github/workflows/win.yml
Comment (nicoddemus) [inline .github/workflows/win.yml:40]: Let's remove this and try again.

Comment (volpatto) [reply .github/workflows/win.yml:40]: OK, I'm on it...

Comment (volpatto) [reply .github/workflows/win.yml]: Didn't work :(

Comment (volpatto) [reply .github/workflows/win.yml]: I will try to set `PYTHONPATH` and `PATH` with `env` field in Tests step.

**Comment by nicoddemus** [comment]
Woot. 😁 

So the problem was that it was using powershell... the fix was to just force GH Actions to use `cmd` with `shell: cmd`. 👍 (Weird, I could swear that was the default).

Thanks @volpatto, LGTM.

(Note we should squash/merge this, I will do it once @prusse-martin approves).

**Review by nicoddemus — APPROVED** [review]

**Comment by volpatto** [comment]
> Woot. 
> 
> So the problem was that it was using powershell... the fix was to just force GH Actions to use `cmd` with `shell: cmd`.  (Weird, I could swear that was the default).
> 
> Thanks @volpatto, LGTM.
> 
> (Note we should squash/merge this, I will do it once @prusse-martin approves).

Yeah, I'm surprised too!

@nicoddemus, BTW, it looks like - at least when I load here - that the tests are not running in Win builds... can you confirm it, please?

**Comment by nicoddemus** [comment]
HMmm yeah it is strange, I don't see either. I will take a look later, I'm on a meeting right now.

File: .github/workflows/linux.yml
Comment (marcelotrevisani) [inline .github/workflows/linux.yml:26]:
```suggestion
        activate-environment: ''
        channels: conda-forge,esss
```

File: .github/workflows/linux.yml
Comment (marcelotrevisani) [inline .github/workflows/linux.yml:34]:
```suggestion
```

File: .github/workflows/linux.yml
Comment (volpatto) [reply .github/workflows/linux.yml:26]: Ah, good suggestions! Thanks!

File: .github/workflows/win.yml
Comment (marcelotrevisani) [inline .github/workflows/win.yml:26]:
```suggestion
        activate-environment: ''
        channels: conda-forge,esss
        channel-priority: true
```

File: .github/workflows/win.yml
Comment (marcelotrevisani) [inline .github/workflows/win.yml:33]:
```suggestion
```

File: .github/workflows/win.yml
Comment (marcelotrevisani) [inline .github/workflows/win.yml:25]: I believe you should set it to false, as you are pining conda to ``4.8.3``
```suggestion
        auto-activate-base: false
```

Comment (marcelotrevisani) [reply .github/workflows/win.yml:25]: and o top of that I believe would be better to use the ``miniconda-version`` and set it to ``4.8.3``

Comment (marcelotrevisani) [reply .github/workflows/win.yml:25]:
```suggestion
        auto-activate-base: false
        miniconda-version: "4.8.3"
```

Comment (volpatto) [reply .github/workflows/win.yml:25]: Yes, I agree. Thanks for your suggestions!

Comment (marcelotrevisani) [reply .github/workflows/win.yml:25]: I forgot to mention to also specify ``conda-version`` to guarantee the ``4.8.3`` will be used

File: .github/workflows/win.yml
Comment (volpatto) [inline .github/workflows/win.yml:35]: Now we could do:

```suggestion
        conda install -c conda-forge conda-devenv
```

File: .github/workflows/win.yml
Comment (volpatto) [inline .github/workflows/win.yml:26]:
```suggestion
        miniconda-version: '4.8.3'
        conda-version: '4.8.3'
```

File: .github/workflows/win.yml
Comment (volpatto) [reply .github/workflows/win.yml]: Done. Thanks!

**Comment by volpatto** [comment]
@marcelotrevisani the modifications look good, but now Win build is failing because there is no Miniconda3 version compatible with the specification. It seems as we have to specify that it is a `py37` Miniconda3. Any advice in this regard?

File: .github/workflows/win.yml
Comment (nicoddemus) [reply .github/workflows/win.yml:26]: Any reason to pin to a specific conda version? I would use the latest for this CI.

**Comment by nicoddemus** [comment]
Let's not pin `conda` to any version, let's just use the latest. Also, please bring back `win-py36` which I commented out while testing.

**Comment by volpatto** [comment]
> Let's not pin `conda` to any version, let's just use the latest. Also, please bring back `win-py36` which I commented out while testing.

OK, I'm on it

**Comment by volpatto** [comment]
GH Actions are working properly, but some tests are failing on Windows.

**Comment by nicoddemus** [comment]
Woot! Thanks @volpatto!

**Pull request merged**
